spirv_parser.cpp

```diff
@@ -1291,6 +1291,12 @@ void Parser::parse(const Instruction &instruction)
 		break;
 	}
 
+	case OpEmitVertex:
+	case OpEndPrimitive:
+		current_function->emits_geometry = true;
```

Comment:
This is not the appropriate place to add this analysis. It makes more sense to add this to ::compile() similar to build_function_control_flow_graphs_and_analyze(). If geometry shaders, run an analysis pass for geometry vertex emission and propagate that information.